ssl/quic/quic_impl.c

```diff
@@ -403,12 +458,19 @@ static int blocking_mode(const QUIC_CONNECTION *qc)
 }
 
 /* SSL_tick; ticks the reactor. */
+QUIC_TAKES_LOCK
 int ossl_quic_tick(QUIC_CONNECTION *qc)
 {
-    if (qc->ch == NULL)
+    if (!quic_lock(qc))
```

Comment:
Sometimes we seem to check the return value from quic_lock and sometimes we don't. Seems inconsistent

Reply:
The only case we don't is in `ossl_quic_free`, which can't fail. But as we are using mutex now (which is infallible) we can get rid of the return value anyway. Done.